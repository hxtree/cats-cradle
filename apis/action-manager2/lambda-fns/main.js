"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const createNote_1 = require("./createNote");
const deleteNote_1 = require("./deleteNote");
const getNoteById_1 = require("./getNoteById");
const listNotes_1 = require("./listNotes");
const updateNote_1 = require("./updateNote");
exports.handler = async (event) => {
    switch (event.info.fieldName) {
        case "getNoteById":
            return await (0, getNoteById_1.default)(event.arguments.noteId);
        case "createNote":
            return await (0, createNote_1.default)(event.arguments.note);
        case "listNotes":
            return await (0, listNotes_1.default)();
        case "deleteNote":
            return await (0, deleteNote_1.default)(event.arguments.noteId);
        case "updateNote":
            return await (0, updateNote_1.default)(event.arguments.note);
        default:
            return null;
    }
};
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoibWFpbi5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbIm1haW4udHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7QUFBQSw2Q0FBc0M7QUFDdEMsNkNBQXNDO0FBQ3RDLCtDQUF3QztBQUN4QywyQ0FBb0M7QUFDcEMsNkNBQXNDO0FBYXRDLE9BQU8sQ0FBQyxPQUFPLEdBQUcsS0FBSyxFQUFFLEtBQWtCLEVBQUUsRUFBRTtJQUMzQyxRQUFRLEtBQUssQ0FBQyxJQUFJLENBQUMsU0FBUyxFQUFFO1FBQzFCLEtBQUssYUFBYTtZQUNkLE9BQU8sTUFBTSxJQUFBLHFCQUFXLEVBQUMsS0FBSyxDQUFDLFNBQVMsQ0FBQyxNQUFNLENBQUMsQ0FBQztRQUNyRCxLQUFLLFlBQVk7WUFDYixPQUFPLE1BQU0sSUFBQSxvQkFBVSxFQUFDLEtBQUssQ0FBQyxTQUFTLENBQUMsSUFBSSxDQUFDLENBQUM7UUFDbEQsS0FBSyxXQUFXO1lBQ1osT0FBTyxNQUFNLElBQUEsbUJBQVMsR0FBRSxDQUFDO1FBQzdCLEtBQUssWUFBWTtZQUNiLE9BQU8sTUFBTSxJQUFBLG9CQUFVLEVBQUMsS0FBSyxDQUFDLFNBQVMsQ0FBQyxNQUFNLENBQUMsQ0FBQztRQUNwRCxLQUFLLFlBQVk7WUFDYixPQUFPLE1BQU0sSUFBQSxvQkFBVSxFQUFDLEtBQUssQ0FBQyxTQUFTLENBQUMsSUFBSSxDQUFDLENBQUM7UUFDbEQ7WUFDSSxPQUFPLElBQUksQ0FBQztLQUNuQjtBQUNMLENBQUMsQ0FBQSIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBjcmVhdGVOb3RlIGZyb20gJy4vY3JlYXRlTm90ZSc7XHJcbmltcG9ydCBkZWxldGVOb3RlIGZyb20gJy4vZGVsZXRlTm90ZSc7XHJcbmltcG9ydCBnZXROb3RlQnlJZCBmcm9tICcuL2dldE5vdGVCeUlkJztcclxuaW1wb3J0IGxpc3ROb3RlcyBmcm9tICcuL2xpc3ROb3Rlcyc7XHJcbmltcG9ydCB1cGRhdGVOb3RlIGZyb20gJy4vdXBkYXRlTm90ZSc7XHJcbmltcG9ydCBOb3RlIGZyb20gJy4vTm90ZSc7XHJcblxyXG50eXBlIEFwcFN5bmNFdmVudCA9IHtcclxuICAgaW5mbzoge1xyXG4gICAgIGZpZWxkTmFtZTogc3RyaW5nXHJcbiAgfSxcclxuICAgYXJndW1lbnRzOiB7XHJcbiAgICAgbm90ZUlkOiBzdHJpbmcsXHJcbiAgICAgbm90ZTogTm90ZVxyXG4gIH1cclxufVxyXG5cclxuZXhwb3J0cy5oYW5kbGVyID0gYXN5bmMgKGV2ZW50OkFwcFN5bmNFdmVudCkgPT4ge1xyXG4gICAgc3dpdGNoIChldmVudC5pbmZvLmZpZWxkTmFtZSkge1xyXG4gICAgICAgIGNhc2UgXCJnZXROb3RlQnlJZFwiOlxyXG4gICAgICAgICAgICByZXR1cm4gYXdhaXQgZ2V0Tm90ZUJ5SWQoZXZlbnQuYXJndW1lbnRzLm5vdGVJZCk7XHJcbiAgICAgICAgY2FzZSBcImNyZWF0ZU5vdGVcIjpcclxuICAgICAgICAgICAgcmV0dXJuIGF3YWl0IGNyZWF0ZU5vdGUoZXZlbnQuYXJndW1lbnRzLm5vdGUpO1xyXG4gICAgICAgIGNhc2UgXCJsaXN0Tm90ZXNcIjpcclxuICAgICAgICAgICAgcmV0dXJuIGF3YWl0IGxpc3ROb3RlcygpO1xyXG4gICAgICAgIGNhc2UgXCJkZWxldGVOb3RlXCI6XHJcbiAgICAgICAgICAgIHJldHVybiBhd2FpdCBkZWxldGVOb3RlKGV2ZW50LmFyZ3VtZW50cy5ub3RlSWQpO1xyXG4gICAgICAgIGNhc2UgXCJ1cGRhdGVOb3RlXCI6XHJcbiAgICAgICAgICAgIHJldHVybiBhd2FpdCB1cGRhdGVOb3RlKGV2ZW50LmFyZ3VtZW50cy5ub3RlKTtcclxuICAgICAgICBkZWZhdWx0OlxyXG4gICAgICAgICAgICByZXR1cm4gbnVsbDtcclxuICAgIH1cclxufSJdfQ==