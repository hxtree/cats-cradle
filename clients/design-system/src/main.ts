export * from './components';
export * from './game-assets';
export * from './types';
export * from './utils';
