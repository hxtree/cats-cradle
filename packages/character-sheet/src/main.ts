import {AppSyncResolverEvent} from 'aws-lambda';
import {VallonOni} from './archetype/npc';

async function getCharacterSheet(id: string): Promise<any> {
  try {
    // const archetype = new VallonOni();
    const character = {
      id: '2222',
      name: 'Vallon Oni',
      description: 'White hair',
    };
    return await Promise.resolve(character);
  } catch (err) {
    Promise.reject(new Error('Failed to get CharacterSheet'));
  }
}

async function listCharacterSheets(): Promise<any> {
  try {
    const archetype = new VallonOni();
    const character = {
      id: '22',
      name: archetype.name,
      description: archetype.description,
    };
    return await Promise.resolve(character);
  } catch (err) {
    Promise.reject(new Error('Failed to get CharacterSheets'));
  }
}

export const handler = async (event: AppSyncResolverEvent<any>) => {
  switch (event.info.fieldName) {
    case 'listCharacterSheets':
      return await listCharacterSheets();
    case 'getCharacterSheetById':
      return await getCharacterSheet(event.arguments.id);
    default:
      return null;
  }
};
