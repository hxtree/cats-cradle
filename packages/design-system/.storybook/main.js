module.exports = {
  stories: ['../src/**/*.stories.tsx'],
  addons: ['@storybook/addon-docs', '@storybook/addon-a11y'],

  framework: '@storybook/react',
  core: {
    builder: 'webpack5',
  },
};
