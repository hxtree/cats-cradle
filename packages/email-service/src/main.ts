// used in development
import { writeFileSync } from 'fs';
import { NestFactory } from '@nestjs/core';
import { SwaggerModule, DocumentBuilder } from '@nestjs/swagger';
import { AppModule } from './app.module';
import { ValidationPipe, DefaultValuePipe } from '@nestjs/common';

async function bootstrap() {
  const app = await NestFactory.create(AppModule);

  const config = new DocumentBuilder()
    .setTitle('@cats-cradles/email-service')
    .setDescription('An API for the email service')
    .setVersion('1.0')
    .build();
  const document = SwaggerModule.createDocument(app, config);
  SwaggerModule.setup('api', app, document);

  // update the openapi-spec
  writeFileSync('./openapi-spec.json', JSON.stringify(document));

  app.useGlobalPipes(
    new ValidationPipe({
      transform: true,
      transformOptions: { enableImplicitConversion: true },
    }),
  );

  await app.listen(3000);
}
bootstrap();
