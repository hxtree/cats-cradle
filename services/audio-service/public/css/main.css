body {
  background-color: #0d6efd;
  font-family: amiga;
  font-size: 1.5em;
  font-family: 'VT323', monospace;
}

#content {
  position: relative;
  top: -45em;
}

#headline {
  position: relative;
  top: 0.5em;
  font-size: 5em;
  background-color: white;
  padding-left: 0.5em;
  text-align: center;
}

#controls {
  float: right;
}

#info {
  background-color: rgba(0, 0, 0, 0.8);
  color: white;
}

.control {
  width: 3em;
  height: 3em;
  background-color: rgba(0, 0, 0, 0.8);
  display: inline-block;
  cursor: pointer;
  margin: 4px;
}

.control:active,
.control.pressed {
  box-shadow: 0px 3px 0px rgba(50, 50, 50, 1), 0px 3px 6px rgba(0, 0, 0, 0.7);
}

#play {
  background-image: url('../img/play.svg');
  background-size: 61%;
  background-position: 16px 9px;
  background-repeat: no-repeat;
}

#pause {
  background-image: url('../img/pause.svg');
  background-size: 61%;
  background-position: center;
  background-repeat: no-repeat;
}

#loop {
  display: none;
}

#loop + label {
  background-color: burlywood;
  background-image: url('../img/loop.svg');
  background-size: 61%;
  background-position: center;
  background-repeat: no-repeat;
}

#loop:checked + label {
  box-shadow: 0px 3px 0px rgba(50, 50, 50, 1), 0px 3px 6px rgba(0, 0, 0, 0.7);
}
