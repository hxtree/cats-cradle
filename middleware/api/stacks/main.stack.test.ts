import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { MainStack } from './main.stack';

describe('MainStack', () => {
  it('should match snapshot test', () => {
    const app = new cdk.App();
    const stack = new MainStack(app, 'MyTestStack');
    const template = Template.fromStack(stack);

    // TODO consider reworking to use DNS instead of incrementing and deploying each time
    const customSnapshotSerializer = {
      test: (val: any) => typeof val === 'string',
      print: (val: any) => {
        let transformedVal = val.replace(/.*?\.zip/g, 'VARIABLE_HASH.zip');
        transformedVal = transformedVal.replace(
          /deployment\d{8}T\d{9}Z[^:]+/g,
          'deployment_UNIQUE_HASH',
        );
        return transformedVal;
      },
    };
    expect.addSnapshotSerializer(customSnapshotSerializer);

    expect(template).toMatchSnapshot();
  });
});
