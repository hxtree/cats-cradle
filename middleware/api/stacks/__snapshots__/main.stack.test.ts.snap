// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainStack should match snapshot test 1`] = `
{
  Outputs: {
    Endpoint: {
      Value: {
        Fn::Join: [
          ,
          [
            https://,
            {
              Ref: MyTestStackdnsdomainMyTestStackdnsdomaincustomdomainB25EDAC1,
            },
          ],
        ],
      },
    },
    MyTestStackdnsdomainMyTestStackdnsdomainrestapiEndpointF2D9559A: {
      Value: {
        Fn::Join: [
          ,
          [
            https://,
            {
              Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiF4C3ABA3,
            },
            .execute-api.,
            {
              Ref: AWS::Region,
            },
            .,
            {
              Ref: AWS::URLSuffix,
            },
            /,
            {
              Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiDeploymentStageprod1A061321,
            },
            /,
          ],
        ],
      },
    },
  },
  Parameters: {
    BootstrapVersion: {
      Default: /cdk-bootstrap/hnb659fds/version,
      Description: Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip],
      Type: AWS::SSM::Parameter::Value<String>,
    },
    MyTestStackdnsdomainMyTestStackdnsdomainhostedzoneidParameter6C99EE2E: {
      Default: my-domains-hosted-zone-id,
      Type: AWS::SSM::Parameter::Value<String>,
    },
    MyTestStackdnsdomainMyTestStackdnsdomainorgformationcertswildcardcert1arnParameter1B552984: {
      Default: orgformation-certs-wildcard-cert1-arn,
      Type: AWS::SSM::Parameter::Value<String>,
    },
    MyTestStackdnsdomainMyTestStackdnsdomainssmdomainnameParameter504B2E51: {
      Default: DOMAIN_NAME,
      Type: AWS::SSM::Parameter::Value<String>,
    },
    lambdalayernestjslatestversionssmParameter: {
      Default: LAMBDA_LAYER_VERSION_NESTJS,
      Type: AWS::SSM::Parameter::Value<String>,
    },
    mongodatabasepasswordssmParameter: {
      Default: MONGO_DATABASE_PASSWORD,
      Type: AWS::SSM::Parameter::Value<String>,
    },
    mongodatabaseurissmParameter: {
      Default: MONGO_DATABASE_URI,
      Type: AWS::SSM::Parameter::Value<String>,
    },
    mongodatabaseuserssmParameter: {
      Default: MONGO_DATABASE_USER,
      Type: AWS::SSM::Parameter::Value<String>,
    },
  },
  Resources: {
    LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A: {
      DependsOn: [
        LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB,
        LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB,
      ],
      Properties: {
        Code: {
          S3Bucket: {
            Fn::Sub: cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region},
          },
          S3Key: 4e26bf2d0a26f2097fb2b261f22bb51e3f6b4b52635777b1e54edbd8e2d58c35.zip,
        },
        Handler: index.handler,
        Role: {
          Fn::GetAtt: [
            LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB,
            Arn,
          ],
        },
        Runtime: nodejs18.x,
        Timeout: 900,
      },
      Type: AWS::Lambda::Function,
    },
    LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB: {
      Properties: {
        AssumeRolePolicyDocument: {
          Statement: [
            {
              Action: sts:AssumeRole,
              Effect: Allow,
              Principal: {
                Service: lambda.amazonaws.com,
              },
            },
          ],
          Version: 2012-10-17,
        },
        ManagedPolicyArns: [
          {
            Fn::Join: [
              ,
              [
                arn:,
                {
                  Ref: AWS::Partition,
                },
                :iam::aws:policy/service-role/AWSLambdaBasicExecutionRole,
              ],
            ],
          },
        ],
      },
      Type: AWS::IAM::Role,
    },
    LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB: {
      Properties: {
        PolicyDocument: {
          Statement: [
            {
              Action: [
                logs:PutRetentionPolicy,
                logs:DeleteRetentionPolicy,
              ],
              Effect: Allow,
              Resource: *,
            },
          ],
          Version: 2012-10-17,
        },
        PolicyName: LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB,
        Roles: [
          {
            Ref: LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB,
          },
        ],
      },
      Type: AWS::IAM::Policy,
    },
    MyTestStackdnsdomainMyTestStackdnsdomainapigatewayaliasrecord1E646064: {
      Properties: {
        AliasTarget: {
          DNSName: {
            Fn::GetAtt: [
              MyTestStackdnsdomainMyTestStackdnsdomaincustomdomainB25EDAC1,
              RegionalDomainName,
            ],
          },
          HostedZoneId: {
            Fn::GetAtt: [
              MyTestStackdnsdomainMyTestStackdnsdomaincustomdomainB25EDAC1,
              RegionalHostedZoneId,
            ],
          },
        },
        HostedZoneId: {
          Ref: MyTestStackdnsdomainMyTestStackdnsdomainhostedzoneidParameter6C99EE2E,
        },
        Name: {
          Fn::Join: [
            ,
            [
              api.,
              {
                Ref: MyTestStackdnsdomainMyTestStackdnsdomainssmdomainnameParameter504B2E51,
              },
              .,
            ],
          ],
        },
        Type: A,
      },
      Type: AWS::Route53::RecordSet,
    },
    MyTestStackdnsdomainMyTestStackdnsdomainapistageC0477553: {
      DependsOn: [
        MyTestStackdnsdomainMyTestStackdnsdomainrestapiAccount8D836574,
      ],
      Properties: {
        DeploymentId: {
          Ref: MyTestStackdnsdomainMyTestStackdnsdomaindeployment_UNIQUE_HASH,
        },
        RestApiId: {
          Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiF4C3ABA3,
        },
        StageName: default,
      },
      Type: AWS::ApiGateway::Stage,
    },
    MyTestStackdnsdomainMyTestStackdnsdomainapisvcdomainname27A6A141: {
      Properties: {
        Description: api service domain name,
        Name: API_SVC_DOMAIN_NAME,
        Type: String,
        Value: {
          Ref: MyTestStackdnsdomainMyTestStackdnsdomaincustomdomainB25EDAC1,
        },
      },
      Type: AWS::SSM::Parameter,
    },
    MyTestStackdnsdomainMyTestStackdnsdomaincustomdomainB25EDAC1: {
      Properties: {
        DomainName: {
          Fn::Join: [
            ,
            [
              api.,
              {
                Ref: MyTestStackdnsdomainMyTestStackdnsdomainssmdomainnameParameter504B2E51,
              },
            ],
          ],
        },
        EndpointConfiguration: {
          Types: [
            REGIONAL,
          ],
        },
        RegionalCertificateArn: {
          Ref: MyTestStackdnsdomainMyTestStackdnsdomainorgformationcertswildcardcert1arnParameter1B552984,
        },
        SecurityPolicy: TLS_1_2,
      },
      Type: AWS::ApiGateway::DomainName,
    },
    MyTestStackdnsdomainMyTestStackdnsdomaincustomdomainMapMyTestStackMyTestStackdnsdomainMyTestStackdnsdomainrestapiE6EBAD2957CC8CE6: {
      Properties: {
        DomainName: {
          Ref: MyTestStackdnsdomainMyTestStackdnsdomaincustomdomainB25EDAC1,
        },
        RestApiId: {
          Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiF4C3ABA3,
        },
        Stage: {
          Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiDeploymentStageprod1A061321,
        },
      },
      Type: AWS::ApiGateway::BasePathMapping,
    },
    MyTestStackdnsdomainMyTestStackdnsdomaindeployment_UNIQUE_HASH: {
      DependsOn: [
        MyTestStackdnsdomainMyTestStackdnsdomainrestapiproxyANY74B2D563,
        MyTestStackdnsdomainMyTestStackdnsdomainrestapiANY413B061D,
      ],
      Properties: {
        RestApiId: {
          Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiF4C3ABA3,
        },
      },
      Type: AWS::ApiGateway::Deployment,
    },
    MyTestStackdnsdomainMyTestStackdnsdomainrestapiANY413B061D: {
      Properties: {
        AuthorizationType: NONE,
        HttpMethod: ANY,
        Integration: {
          IntegrationHttpMethod: POST,
          Type: AWS_PROXY,
          Uri: {
            Fn::Join: [
              ,
              [
                arn:,
                {
                  Ref: AWS::Partition,
                },
                :apigateway:,
                {
                  Ref: AWS::Region,
                },
                :lambda:path/2015-03-31/functions/,
                {
                  Fn::GetAtt: [
                    apiservicestacknestjsNodeJsLambdaA2C684BA,
                    Arn,
                  ],
                },
                /invocations,
              ],
            ],
          },
        },
        ResourceId: {
          Fn::GetAtt: [
            MyTestStackdnsdomainMyTestStackdnsdomainrestapiF4C3ABA3,
            RootResourceId,
          ],
        },
        RestApiId: {
          Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiF4C3ABA3,
        },
      },
      Type: AWS::ApiGateway::Method,
    },
    MyTestStackdnsdomainMyTestStackdnsdomainrestapiANYApiPermissionMyTestStackMyTestStackdnsdomainMyTestStackdnsdomainrestapiE6EBAD29ANYB678C578: {
      Properties: {
        Action: lambda:InvokeFunction,
        FunctionName: {
          Fn::GetAtt: [
            apiservicestacknestjsNodeJsLambdaA2C684BA,
            Arn,
          ],
        },
        Principal: apigateway.amazonaws.com,
        SourceArn: {
          Fn::Join: [
            ,
            [
              arn:,
              {
                Ref: AWS::Partition,
              },
              :execute-api:,
              {
                Ref: AWS::Region,
              },
              :,
              {
                Ref: AWS::AccountId,
              },
              :,
              {
                Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiF4C3ABA3,
              },
              /,
              {
                Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiDeploymentStageprod1A061321,
              },
              /*/,
            ],
          ],
        },
      },
      Type: AWS::Lambda::Permission,
    },
    MyTestStackdnsdomainMyTestStackdnsdomainrestapiANYApiPermissionTestMyTestStackMyTestStackdnsdomainMyTestStackdnsdomainrestapiE6EBAD29ANY7EC26F97: {
      Properties: {
        Action: lambda:InvokeFunction,
        FunctionName: {
          Fn::GetAtt: [
            apiservicestacknestjsNodeJsLambdaA2C684BA,
            Arn,
          ],
        },
        Principal: apigateway.amazonaws.com,
        SourceArn: {
          Fn::Join: [
            ,
            [
              arn:,
              {
                Ref: AWS::Partition,
              },
              :execute-api:,
              {
                Ref: AWS::Region,
              },
              :,
              {
                Ref: AWS::AccountId,
              },
              :,
              {
                Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiF4C3ABA3,
              },
              /test-invoke-stage/*/,
            ],
          ],
        },
      },
      Type: AWS::Lambda::Permission,
    },
    MyTestStackdnsdomainMyTestStackdnsdomainrestapiAccount8D836574: {
      DeletionPolicy: Retain,
      DependsOn: [
        MyTestStackdnsdomainMyTestStackdnsdomainrestapiF4C3ABA3,
      ],
      Properties: {
        CloudWatchRoleArn: {
          Fn::GetAtt: [
            MyTestStackdnsdomainMyTestStackdnsdomainrestapiCloudWatchRole576320DC,
            Arn,
          ],
        },
      },
      Type: AWS::ApiGateway::Account,
      UpdateReplacePolicy: Retain,
    },
    MyTestStackdnsdomainMyTestStackdnsdomainrestapiCloudWatchRole576320DC: {
      DeletionPolicy: Retain,
      Properties: {
        AssumeRolePolicyDocument: {
          Statement: [
            {
              Action: sts:AssumeRole,
              Effect: Allow,
              Principal: {
                Service: apigateway.amazonaws.com,
              },
            },
          ],
          Version: 2012-10-17,
        },
        ManagedPolicyArns: [
          {
            Fn::Join: [
              ,
              [
                arn:,
                {
                  Ref: AWS::Partition,
                },
                :iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs,
              ],
            ],
          },
        ],
      },
      Type: AWS::IAM::Role,
      UpdateReplacePolicy: Retain,
    },
    MyTestStackdnsdomainMyTestStackdnsdomainrestapiDeployment2879F381befc6bceb2ba77e261b13d1c0830c668: {
      DependsOn: [
        MyTestStackdnsdomainMyTestStackdnsdomainrestapiproxyANY74B2D563,
        MyTestStackdnsdomainMyTestStackdnsdomainrestapiproxy3260E5C6,
        MyTestStackdnsdomainMyTestStackdnsdomainrestapiANY413B061D,
      ],
      Properties: {
        Description: api Service API Gateway,
        RestApiId: {
          Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiF4C3ABA3,
        },
      },
      Type: AWS::ApiGateway::Deployment,
    },
    MyTestStackdnsdomainMyTestStackdnsdomainrestapiDeploymentStageprod1A061321: {
      DependsOn: [
        MyTestStackdnsdomainMyTestStackdnsdomainrestapiAccount8D836574,
      ],
      Properties: {
        DeploymentId: {
          Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiDeployment2879F381befc6bceb2ba77e261b13d1c0830c668,
        },
        MethodSettings: [
          {
            DataTraceEnabled: false,
            HttpMethod: *,
            ResourcePath: /*,
            ThrottlingBurstLimit: 10,
            ThrottlingRateLimit: 10,
          },
        ],
        RestApiId: {
          Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiF4C3ABA3,
        },
        StageName: prod,
      },
      Type: AWS::ApiGateway::Stage,
    },
    MyTestStackdnsdomainMyTestStackdnsdomainrestapiF4C3ABA3: {
      Properties: {
        Description: api Service API Gateway,
        Name: api RestAPI,
      },
      Type: AWS::ApiGateway::RestApi,
    },
    MyTestStackdnsdomainMyTestStackdnsdomainrestapiproxy3260E5C6: {
      Properties: {
        ParentId: {
          Fn::GetAtt: [
            MyTestStackdnsdomainMyTestStackdnsdomainrestapiF4C3ABA3,
            RootResourceId,
          ],
        },
        PathPart: {proxy+},
        RestApiId: {
          Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiF4C3ABA3,
        },
      },
      Type: AWS::ApiGateway::Resource,
    },
    MyTestStackdnsdomainMyTestStackdnsdomainrestapiproxyANY74B2D563: {
      Properties: {
        AuthorizationType: NONE,
        HttpMethod: ANY,
        Integration: {
          IntegrationHttpMethod: POST,
          Type: AWS_PROXY,
          Uri: {
            Fn::Join: [
              ,
              [
                arn:,
                {
                  Ref: AWS::Partition,
                },
                :apigateway:,
                {
                  Ref: AWS::Region,
                },
                :lambda:path/2015-03-31/functions/,
                {
                  Fn::GetAtt: [
                    apiservicestacknestjsNodeJsLambdaA2C684BA,
                    Arn,
                  ],
                },
                /invocations,
              ],
            ],
          },
        },
        ResourceId: {
          Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiproxy3260E5C6,
        },
        RestApiId: {
          Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiF4C3ABA3,
        },
      },
      Type: AWS::ApiGateway::Method,
    },
    MyTestStackdnsdomainMyTestStackdnsdomainrestapiproxyANYApiPermissionMyTestStackMyTestStackdnsdomainMyTestStackdnsdomainrestapiE6EBAD29ANYproxyD28F0389: {
      Properties: {
        Action: lambda:InvokeFunction,
        FunctionName: {
          Fn::GetAtt: [
            apiservicestacknestjsNodeJsLambdaA2C684BA,
            Arn,
          ],
        },
        Principal: apigateway.amazonaws.com,
        SourceArn: {
          Fn::Join: [
            ,
            [
              arn:,
              {
                Ref: AWS::Partition,
              },
              :execute-api:,
              {
                Ref: AWS::Region,
              },
              :,
              {
                Ref: AWS::AccountId,
              },
              :,
              {
                Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiF4C3ABA3,
              },
              /,
              {
                Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiDeploymentStageprod1A061321,
              },
              /*/*,
            ],
          ],
        },
      },
      Type: AWS::Lambda::Permission,
    },
    MyTestStackdnsdomainMyTestStackdnsdomainrestapiproxyANYApiPermissionTestMyTestStackMyTestStackdnsdomainMyTestStackdnsdomainrestapiE6EBAD29ANYproxy132638AB: {
      Properties: {
        Action: lambda:InvokeFunction,
        FunctionName: {
          Fn::GetAtt: [
            apiservicestacknestjsNodeJsLambdaA2C684BA,
            Arn,
          ],
        },
        Principal: apigateway.amazonaws.com,
        SourceArn: {
          Fn::Join: [
            ,
            [
              arn:,
              {
                Ref: AWS::Partition,
              },
              :execute-api:,
              {
                Ref: AWS::Region,
              },
              :,
              {
                Ref: AWS::AccountId,
              },
              :,
              {
                Ref: MyTestStackdnsdomainMyTestStackdnsdomainrestapiF4C3ABA3,
              },
              /test-invoke-stage/*/*,
            ],
          ],
        },
      },
      Type: AWS::Lambda::Permission,
    },
    apiservicestacknestjsNodeJsLambdaA2C684BA: {
      DependsOn: [
        apiservicestacknestjsNodeJsLambdaServiceRole7CC8341F,
      ],
      Properties: {
        Code: {
          S3Bucket: {
            Fn::Sub: cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region},
          },
          S3Key: b8ba19c32be3f3a103e1b09b48ddc7264973a45b1bd00c108f1abb1a314e78cb.zip,
        },
        Environment: {
          Variables: {
            API_SVC_DOMAIN_NAME: {{resolve:ssm:/API_SVC_DOMAIN_NAME}},
            AWS_ACCOUNT_ID: {
              Ref: AWS::AccountId,
            },
            CHARACTER_SHEETS_SVC_DOMAIN_NAME: {{resolve:ssm:/CHARACTER_SHEETS_SVC_DOMAIN_NAME}},
            EMAIL_SERVICE_SVC_DOMAIN_NAME: {{resolve:ssm:/EMAIL_SERVICE_SVC_DOMAIN_NAME}},
            HTML_TO_PDF_SVC_DOMAIN_NAME: {{resolve:ssm:/HTML_TO_PDF_SVC_DOMAIN_NAME}},
            INSTANCES_SVC_DOMAIN_NAME: {{resolve:ssm:/INSTANCES_SVC_DOMAIN_NAME}},
            LUCK_BY_DICE_SVC_DOMAIN_NAME: {{resolve:ssm:/LUCK_BY_DICE_SVC_DOMAIN_NAME}},
            MONGO_DATABASE_PASSWORD: {
              Ref: mongodatabasepasswordssmParameter,
            },
            MONGO_DATABASE_URI: {
              Ref: mongodatabaseurissmParameter,
            },
            MONGO_DATABASE_USER: {
              Ref: mongodatabaseuserssmParameter,
            },
            PLAYER_ACHIEVEMENTS_SVC_DOMAIN_NAME: {{resolve:ssm:/PLAYER_ACHIEVEMENTS_SVC_DOMAIN_NAME}},
            STAGE_NAME: default,
            USER_SVC_DOMAIN_NAME: {{resolve:ssm:/USER_SVC_DOMAIN_NAME}},
          },
        },
        Handler: index.handler,
        Layers: [
          {
            Ref: lambdalayernestjslatestversionssmParameter,
          },
        ],
        MemorySize: 1024,
        Role: {
          Fn::GetAtt: [
            apiservicestacknestjsNodeJsLambdaServiceRole7CC8341F,
            Arn,
          ],
        },
        Runtime: nodejs18.x,
        Timeout: 30,
      },
      Type: AWS::Lambda::Function,
    },
    apiservicestacknestjsNodeJsLambdaLogRetention0873DFB6: {
      Properties: {
        LogGroupName: {
          Fn::Join: [
            ,
            [
              /aws/lambda/,
              {
                Ref: apiservicestacknestjsNodeJsLambdaA2C684BA,
              },
            ],
          ],
        },
        RetentionInDays: 1,
        ServiceToken: {
          Fn::GetAtt: [
            LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A,
            Arn,
          ],
        },
      },
      Type: Custom::LogRetention,
    },
    apiservicestacknestjsNodeJsLambdaServiceRole7CC8341F: {
      Properties: {
        AssumeRolePolicyDocument: {
          Statement: [
            {
              Action: sts:AssumeRole,
              Effect: Allow,
              Principal: {
                Service: lambda.amazonaws.com,
              },
            },
          ],
          Version: 2012-10-17,
        },
        ManagedPolicyArns: [
          {
            Fn::Join: [
              ,
              [
                arn:,
                {
                  Ref: AWS::Partition,
                },
                :iam::aws:policy/service-role/AWSLambdaBasicExecutionRole,
              ],
            ],
          },
        ],
      },
      Type: AWS::IAM::Role,
    },
  },
  Rules: {
    CheckBootstrapVersion: {
      Assertions: [
        {
          Assert: {
            Fn::Not: [
              {
                Fn::Contains: [
                  [
                    1,
                    2,
                    3,
                    4,
                    5,
                  ],
                  {
                    Ref: BootstrapVersion,
                  },
                ],
              },
            ],
          },
          AssertDescription: CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.,
        },
      ],
    },
  },
}
`;
